### ISSUE TITLE
Fix vr-xcon busy looping

### ISSUE THREAD

**Issue by plajjan** [plajjan/vrnetlab#182]
This is ridiculous:
```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                                                                                                        
25813 root      20   0   28316   8388   2188 R  94,9  0,0  29:52.14 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.42.1/30 --ipv4-route 192.168.42.2 --ipv6-address 2001:db8:5::42:1/126 --ipv6-route 2001:db8:5::42:2                                                     
23637 root      20   0   28316  10008   3788 R  94,6  0,0   8:53.43 python3 /xcon.py --tap-listen 1 --ipv6-address 2001:db8:5::3:1/126 --ipv6-route 2001:db8:5::3:2                                                                                                                
39727 root      20   0   28252   8820   2652 R  94,1  0,0  12:27.02 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.2.1/30 --ipv4-route 192.168.2.2 --ipv6-address 2001:db8:5::2:1/126 --ipv6-route 2001:db8:5::2:2                                                         
17438 root      20   0   28304  10048   3864 R  93,5  0,0   8:40.97 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.1.1/30 --ipv4-route 192.168.1.2 --ipv6-address 2001:db8:5::1:1/126 --ipv6-route 2001:db8:5::1:2                                                         
14581 root      20   0   28240  10076   3872 R  92,1  0,0   8:44.64 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.42.1/30 --ipv4-route 192.168.42.2 --ipv6-address 2001:db8:5::42:1/126 --ipv6-route 2001:db8:5::42:2                                                     
31279 root      20   0   28304   8348   2172 R  91,8  0,0  30:20.42 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.61.1/30 --ipv4-route 192.168.61.2 --ipv6-address 2001:db8:5::61:1/126 --ipv6-route 2001:db8:5::61:2                                                     
26561 root      20   0   28240   8216   2068 R  91,5  0,0  30:11.62 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.32.1/30 --ipv4-route 192.168.32.2 --ipv6-address 2001:db8:5::32:1/126 --ipv6-route 2001:db8:5::32:2                                                     
11108 root      20   0   28304  10016   3812 R  90,7  0,0   8:46.36 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.41.1/30 --ipv4-route 192.168.41.2 --ipv6-address 2001:db8:5::41:1/126 --ipv6-route 2001:db8:5::41:2                                                     
 8814 root      20   0   28304  10344   4136 R  90,4  0,0   9:02.85 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.32.1/30 --ipv4-route 192.168.32.2 --ipv6-address 2001:db8:5::32:1/126 --ipv6-route 2001:db8:5::32:2                                                     
36976 root      20   0   28304   8316   2108 R  90,4  0,0  29:20.37 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.2.1/30 --ipv4-route 192.168.2.2 --ipv6-address 2001:db8:5::2:1/126 --ipv6-route 2001:db8:5::2:2                                                         
 1072 root      20   0   28304  10340   4132 R  89,9  0,0   6:47.09 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.22.1/30 --ipv4-route 192.168.22.2 --ipv6-address 2001:db8:5::22:1/126 --ipv6-route 2001:db8:5::22:2                                                     
 4038 root      20   0   28320   8524   2484 R  89,9  0,0  30:22.89 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.11.1/30 --ipv4-route 192.168.11.2 --ipv6-address 2001:db8:5::11:1/126 --ipv6-route 2001:db8:5::11:2                                                     
26883 root      20   0   28316  10148   3928 R  89,0  0,0   8:46.85 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.11.1/30 --ipv4-route 192.168.11.2 --ipv6-address 2001:db8:5::11:1/126 --ipv6-route 2001:db8:5::11:2                                                     
 6776 root      20   0   28304   8232   2356 R  88,7  0,0  24:25.25 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.22.1/30 --ipv4-route 192.168.22.2 --ipv6-address 2001:db8:5::22:1/126 --ipv6-route 2001:db8:5::22:2                                                     
 7409 root      20   0   28308   8724   2592 R  87,9  0,0  12:57.15 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.11.1/30 --ipv4-route 192.168.11.2 --ipv6-address 2001:db8:5::11:1/126 --ipv6-route 2001:db8:5::11:2                                                     
17021 root      20   0   28308   8428   2240 R  87,9  0,0  30:20.60 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.51.1/30 --ipv4-route 192.168.51.2 --ipv6-address 2001:db8:5::51:1/126 --ipv6-route 2001:db8:5::51:2                                                     
21371 root      20   0   28304   8460   2260 R  86,2  0,0  29:43.11 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.31.1/30 --ipv4-route 192.168.31.2 --ipv6-address 2001:db8:5::31:1/126 --ipv6-route 2001:db8:5::31:2                                                     
  315 root      20   0   28240   8116   2048 R  85,4  0,0  29:43.56 python3 /xcon.py --tap-listen 1 --ipv6-address 2001:db8:5::3:1/126 --ipv6-route 2001:db8:5::3:2                                                                                                                
29483 root      20   0   28304   9040   2880 R  85,1  0,0  17:29.36 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.31.1/30 --ipv4-route 192.168.31.2 --ipv6-address 2001:db8:5::31:1/126 --ipv6-route 2001:db8:5::31:2                                                     
11759 root      20   0   28304   8752   2588 R  84,2  0,0  12:24.34 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.12.1/30 --ipv4-route 192.168.12.2 --ipv6-address 2001:db8:5::12:1/126 --ipv6-route 2001:db8:5::12:2                                                     
 4232 root      20   0   28304  10008   3856 R  82,3  0,0   6:18.87 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.23.1/30 --ipv4-route 192.168.23.2 --ipv6-address 2001:db8:5::23:1/126 --ipv6-route 2001:db8:5::23:2                                                     
11740 root      20   0   28304   8836   2644 R  74,9  0,0  24:23.53 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.23.1/30 --ipv4-route 192.168.23.2 --ipv6-address 2001:db8:5::23:1/126 --ipv6-route 2001:db8:5::23:2                                                     
28236 root      20   0   28240   8872   2736 R  73,0  0,0  12:48.65 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.42.1/30 --ipv4-route 192.168.42.2 --ipv6-address 2001:db8:5::42:1/126 --ipv6-route 2001:db8:5::42:2                                                     
 2686 root      20   0   28316   8400   2324 R  72,7  0,0  24:38.62 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.21.1/30 --ipv4-route 192.168.21.2 --ipv6-address 2001:db8:5::21:1/126 --ipv6-route 2001:db8:5::21:2                                                     
16585 root      20   0   28316   8396   2232 R  72,4  0,0  23:33.17 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.24.1/30 --ipv4-route 192.168.24.2 --ipv6-address 2001:db8:5::24:1/126 --ipv6-route 2001:db8:5::24:2                                                     
29861 root      20   0   28320  10008   3816 R  67,0  0,0   6:55.20 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.12.1/30 --ipv4-route 192.168.12.2 --ipv6-address 2001:db8:5::12:1/126 --ipv6-route 2001:db8:5::12:2                                                     
 5245 root      20   0   28304  10104   3900 R  65,6  0,0   6:46.81 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.31.1/30 --ipv4-route 192.168.31.2 --ipv6-address 2001:db8:5::31:1/126 --ipv6-route 2001:db8:5::31:2                                                     
14889 root      20   0   28616  10940   4856 R  65,6  0,0   3:14.40 python3 /xcon.py --tap-listen 2 --ipv6-address 2003:1f0b:fffe:1::2/64 --ipv6-route 2003:1f0b:fffe:1::1 --tap-if tap1                                                                                           
12795 root      20   0   28616  10436   4864 R  58,6  0,0   3:00.88 python3 /xcon.py --tap-listen 1 --ipv6-address 2003:1b3b:fffe:1::2/64                                                                                                                                          
19206 root      20   0   28304   8888   2708 R  57,2  0,0  12:55.38 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.24.1/30 --ipv4-route 192.168.24.2 --ipv6-address 2001:db8:5::24:1/126 --ipv6-route 2001:db8:5::24:2                                                     
19668 root      20   0 18,452g 5,658g   5116 S  56,9  2,2  18:57.33 qemu-system-x86_64 -enable-kvm -display none -machine pc -monitor tcp:0.0.0.0:4000,server,nowait -m 16384 -serial telnet:0.0.0.0:5000,server,nowait -drive if=ide,file=/Simulator_V100R001C00SPC002T-overlay.+ 
 7003 root      20   0   28240   8628   2628 R  56,6  0,0  12:55.40 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.22.1/30 --ipv4-route 192.168.22.2 --ipv6-address 2001:db8:5::22:1/126 --ipv6-route 2001:db8:5::22:2                                                     
30411 root      20   0 8440012 3,997g   3372 R  53,2  1,6  14:41.05 qemu-system-x86_64 -enable-kvm -display none -machine pc -monitor tcp:0.0.0.0:4001,server,nowait -m 4096 -serial telnet:0.0.0.0:5001,server,nowait -drive if=ide,file=/vmx/vfpc-overlay.img -cpu SandyBridge + 
20781 root      20   0   28252  10096   3876 R  53,0  0,0   6:44.40 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.2.1/30 --ipv4-route 192.168.2.2 --ipv6-address 2001:db8:5::2:1/126 --ipv6-route 2001:db8:5::2:2                                                         
 8901 root      20   0   28304   8492   2332 R  50,7  0,0  24:44.63 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.12.1/30 --ipv4-route 192.168.12.2 --ipv6-address 2001:db8:5::12:1/126 --ipv6-route 2001:db8:5::12:2                                                     
 2687 root      20   0   28308   8904   2700 R  50,1  0,0  12:39.10 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.21.1/30 --ipv4-route 192.168.21.2 --ipv6-address 2001:db8:5::21:1/126 --ipv6-route 2001:db8:5::21:2                                                     
 3158 root      20   0   28304   8988   2852 R  49,3  0,0  15:34.02 python3 /xcon.py --tap-listen 1 --ipv6-address 2001:db8:5::3:1/126 --ipv6-route 2001:db8:5::3:2                                                                                                                
 3493 root      20   0 8727856 4,347g   3928 R  49,0  1,7  14:17.32 qemu-system-x86_64 -enable-kvm -display none -machine pc -monitor tcp:0.0.0.0:4001,server,nowait -m 4096 -serial telnet:0.0.0.0:5001,server,nowait -drive if=ide,file=/vmx/vfpc-overlay.img -cpu SandyBridge + 
30578 root      20   0   28304   8552   2356 R  49,0  0,0  24:29.95 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.1.1/30 --ipv4-route 192.168.1.2 --ipv6-address 2001:db8:5::1:1/126 --ipv6-route 2001:db8:5::1:2                                                         
34554 root      20   0   28304   8792   2604 R  48,5  0,0  12:48.75 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.32.1/30 --ipv4-route 192.168.32.2 --ipv6-address 2001:db8:5::32:1/126 --ipv6-route 2001:db8:5::32:2                                                     
39285 root      20   0   28316  10168   3944 R  48,2  0,0   6:31.03 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.51.1/30 --ipv4-route 192.168.51.2 --ipv6-address 2001:db8:5::51:1/126 --ipv6-route 2001:db8:5::51:2                                                     
23340 root      20   0   28308   8796   2616 R  47,9  0,0  12:26.49 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.41.1/30 --ipv4-route 192.168.41.2 --ipv6-address 2001:db8:5::41:1/126 --ipv6-route 2001:db8:5::41:2                                                     
 7400 root      20   0   28304  10024   3816 R  47,6  0,0   6:51.38 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.24.1/30 --ipv4-route 192.168.24.2 --ipv6-address 2001:db8:5::24:1/126 --ipv6-route 2001:db8:5::24:2                                                     
14701 root      20   0 19,522g 5,631g   5372 S  47,0  2,2  19:11.60 qemu-system-x86_64 -enable-kvm -display none -machine pc -monitor tcp:0.0.0.0:4000,server,nowait -m 16384 -serial telnet:0.0.0.0:5000,server,nowait -drive if=ide,file=/Simulator_V100R001C00SPC002T-overlay.+ 
37408 root      20   0   28240   9820   3712 R  46,8  0,0   6:37.15 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.21.1/30 --ipv4-route 192.168.21.2 --ipv6-address 2001:db8:5::21:1/126 --ipv6-route 2001:db8:5::21:2                                                     
20121 root      20   0   28304   9060   2932 R  46,5  0,0  15:55.19 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.51.1/30 --ipv4-route 192.168.51.2 --ipv6-address 2001:db8:5::51:1/126 --ipv6-route 2001:db8:5::51:2                                                     
 1629 root      20   0   28304  10324   4116 R  44,5  0,0   6:43.97 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.61.1/30 --ipv4-route 192.168.61.2 --ipv6-address 2001:db8:5::61:1/126 --ipv6-route 2001:db8:5::61:2                                                     
24684 root      20   0   28304   8976   2820 R  44,5  0,0  13:03.20 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.61.1/30 --ipv4-route 192.168.61.2 --ipv6-address 2001:db8:5::61:1/126 --ipv6-route 2001:db8:5::61:2                                                     
36208 root      20   0   28304   8784   2596 R  43,4  0,0  12:51.82 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.1.1/30 --ipv4-route 192.168.1.2 --ipv6-address 2001:db8:5::1:1/126 --ipv6-route 2001:db8:5::1:2                                                         
20811 root      20   0   28308   8536   2364 R  42,8  0,0  24:17.14 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.41.1/30 --ipv4-route 192.168.41.2 --ipv6-address 2001:db8:5::41:1/126 --ipv6-route 2001:db8:5::41:2                                                     
11765 root      20   0   28308   8792   2700 R  42,0  0,0  12:42.29 python3 /xcon.py --tap-listen 1 --ipv4-address 192.168.23.1/30 --ipv4-route 192.168.23.2 --ipv6-address 2001:db8:5::23:1/126 --ipv6-route 2001:db8:5::23:2               
```
The top CPU hoggers are all vr-xcon processes and not virtual routers. Liek wtf. fix omgwtfbbqplxz

**Comment by plajjan** [comment]
Is this a job for asyncio?

**Comment by plajjan** [comment]
Fixed with 043c0809fef. Closing.
